slo.go

```diff
@@ -77,32 +105,55 @@ type Query struct {
 	Type      QueryType       `json:"type"`
 }
 
+// QueryType defines model for Query.Type.
 type QueryType string
 
+// RatioQuery defines model for RatioQuery.
 type RatioQuery struct {
 	GroupByLabels []string  `json:"groupByLabels,omitempty"`
 	SuccessMetric MetricDef `json:"successMetric"`
 	TotalMetric   MetricDef `json:"totalMetric"`
 }
 
-type Slo struct {
-	Alerting              *Alerting     `json:"alerting,omitempty"`
-	Description           string        `json:"description"`
+// ReadOnly defines model for ReadOnly.
+type ReadOnly struct {
 	DrillDownDashboardRef *DashboardRef `json:"drillDownDashboardRef,omitempty"`
-	Labels                []Label       `json:"labels,omitempty"`
-	Name                  string        `json:"name"`
-	Objectives            []Objective   `json:"objectives"`
-	Query                 Query         `json:"query"`
-	UUID                  string        `json:"uuid"`
+	Provenance            *string       `json:"provenance,omitempty"`
```

Comment:
nit - you're not advocating for pointer, you're just taking this `*string` type from the cue-generated structs, yes?

Since this is a library for public consumption, it seems perhaps better to follow idomatic golang and just make it a `string` since we're not doing anything fancy that requires the pointer type.